src/test/java/org/junitpioneer/jupiter/resource/ResourcesTests.java

```diff
@@ -0,0 +1,543 @@
+						.executeTestClass(ThrowOnSharedRFCreateTestCases.class);
+				executionResults
+						.allEvents()
+						.debug()
+						.assertThatEvents()
+						.haveExactly(//
+							1, //
+							finished(//
+								throwable(//
+									instanceOf(ParameterResolutionException.class), //
+									message("Unable to create a resource from `" + ThrowOnRFCreateResourceFactory.class
+											+ "`"), //
+									cause(//
+										instanceOf(EXPECTED_THROW_ON_RF_CREATE_EXCEPTION.getClass()), //
+										message(EXPECTED_THROW_ON_RF_CREATE_EXCEPTION.getMessage())))));
+			}
+
+		}
+
+		@DisplayName("and the factory throws on ::close")
+		@Nested
+		class AndFactoryThrowsOnCloseTests {
+
+			@DisplayName("then the thrown exception is propagated")
+			@Test
+			void thenThrownExceptionIsPropagated() {
+				ExecutionResults executionResults = PioneerTestKit
+						.executeTestClass(ThrowOnSharedRFCloseTestCases.class);
+				executionResults
+						.allEvents()
+						.debug()
+						.assertThatEvents()
+						.haveExactly(//
+							1, //
+							finished(//
+								throwable(//
+									instanceOf(EXPECTED_THROW_ON_RF_CLOSE_EXCEPTION.getClass()), //
+									message(EXPECTED_THROW_ON_RF_CLOSE_EXCEPTION.getMessage()))));
+			}
+
+		}
+
+		@DisplayName("and a resource is created")
+		@Nested
+		class AndResourceIsCreatedTests {
+
+			@DisplayName("and the resource throws on ::get")
+			@Nested
+			class AndResourceThrowsOnGetTests {
+
+				@DisplayName("then the thrown exception is wrapped and propagated")
+				@Test
+				void thenThrownExceptionIsWrappedAndPropagated() {
+					ExecutionResults executionResults = PioneerTestKit
+							.executeTestClass(ThrowOnSharedRGetTestCases.class);
+					executionResults
+							.allEvents()
+							.debug()
+							.assertThatEvents()
+							.haveExactly(//
+								1, //
+								finished(//
+									throwable(//
+										instanceOf(ParameterResolutionException.class), //
+										message("Unable to get the contents of the resource created by `"
+												+ ThrowOnRGetResourceFactory.class + "`"), //
+										cause(//
+											instanceOf(EXPECTED_THROW_ON_R_GET_EXCEPTION.getClass()), //
+											message(EXPECTED_THROW_ON_R_GET_EXCEPTION.getMessage())))));
+				}
+
+			}
+
+			@DisplayName("and the resource throws on ::close")
+			@Nested
+			class AndResourceThrowsOnCloseTests {
+
+				@DisplayName("then the thrown exception is propagated")
+				@Test
+				void thenThrownExceptionIsWrappedAndPropagated() {
+					ExecutionResults executionResults = PioneerTestKit
+							.executeTestClass(ThrowOnSharedRCloseTestCases.class);
+					executionResults
+							.allEvents()
+							.debug()
+							.assertThatEvents()
+							.haveExactly(//
+								1, //
+								finished(//
+									throwable(//
+										instanceOf(EXPECTED_THROW_ON_R_CLOSE_EXCEPTION.getClass()), //
+										message(EXPECTED_THROW_ON_R_CLOSE_EXCEPTION.getMessage()))));
+				}
+
+			}
+
+		}
+
```

Comment:
* assert that for every `@Shared` with a new name, `ResourceFactory::create` is called
* assert that for every `@Shared` with a known name, `ResourceFactory::create` is not called

Reply:
@nipafx Sorry for the hiatus, I've just acted upon this feedback! See commit https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/5ab5f3862ff22ec98f4f2d89e1e29694c38b3ab1.